src/ts/components/dashboard/Dashboard.tsx

```diff
@@ -18,15 +33,43 @@ class Dashboard extends React.Component<IDashboardProps> {
     })
   }
 
+  shortAddress = (address) => {
+    return address.substring(0, 5) + '...' + address.substring(address.length - 10)
+  }
+
+  loadAccounts = () => {
+    getAccounts().then(
+      result => {
+        console.log(`get account ${result}`)
+        if (!Array.isArray(result) || !result.length) {
+          return
+        }
+        this.props.setCurrentAddressAndName(result[0].address, result[0].meta.name)
+        this.setState({
+          initializing: false,
+          currentAddress: result[0].address,
+          currentName: result[0].meta.name
+        })
+      }
+    )
+  }
+
+  componentWillMount () {
+    this.loadAccounts()
+  }
+
   render () {
+    if (this.state.initializing) {
+      return (null)
+    }
     return (
       <ContentContainer>
         <Section>
-          <Title>
-            Dashboard goes here
-          </Title>
+          Account Name : {this.state.currentName}
```

Comment:
can we add i18n message key here to save refactoring later?

Reply:
updated